### ISSUE TITLE
Duplicate --bes_header causes Bazel to exit with code 37 without output

### ISSUE THREAD

**Issue by jayconrod** [bazelbuild/bazel#20681]
### Description of the bug:

When `--bes_header` is used to set the same header more than once, Bazel crashes with no output, exit code 37 (internal error).

There's no real need to set the same header more than once, but it's easy to do accidentally with a redundant `--config` somewhere.

### Which category does this issue belong to?

Remote Execution

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

1. Checkout https://github.com/EngFlow/example
2. Create `.bazelrc.user`:

```
build:opal_auth --bes_header=x-redundant=1
build:opal_auth --bes_header=x-redundant=1
```

3. `bazel build --config=opal --config=remote_linux_x64 //cpp`

### Which operating system are you running Bazel on?

Linux / x86_64, also reproduced on macOS / arm64

### What is the output of `bazel info release`?

release 8.0.0-pre.20231030.2, also reproduced on 6.4.0

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

n/a

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

```text
n/a
```

### Is this a regression? If yes, please try to identify the Bazel commit where the bug was introduced.

Not a recent one.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!

**Comment by jayconrod** [comment]
Verified fixed!
